### ISSUE TITLE
BPython crashing when typing in a numpy array

### ISSUE THREAD

**Issue by lukik** [bpython/bpython#983]
Getting an error when I type in a numpy array using BPython. 

When using default Python this error doesn't come up. And typing in a normal Python list also doesn't error out. If I however copy paste the array in full it doesn't error out. Only crashes when I start typing it out digit by digit.

Sample array: ar = np.array([1,2,3,4,5,6])

It crashes the moment I type in 2.

```
import numpy as np
ar = np.array([1,2


Traceback (most recent call last):
  File "/home/lukik/anaconda3/envs/ds_env/bin/bpython", line 11, in <module>
	sys.exit(main())
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsies.py", line 221, in main
	exit_value = repl.mainloop(True, paste)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsies.py", line 133, in mainloop
	self.process_event_and_paint(e)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsies.py", line 97, in process_event_and_paint
	self.process_event(e)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsiesfrontend/repl.py", line 658, in process_event
	return self.process_key_event(e)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsiesfrontend/repl.py", line 825, in process_key_event
	self.add_normal_character(e)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsiesfrontend/repl.py", line 1111, in add_normal_character
	self.cursor_offset += 1
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsiesfrontend/repl.py", line 1821, in _set_cursor_offset
	self.update_completion()
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/curtsiesfrontend/repl.py", line 1143, in update_completion
	self.list_win_visible = self.complete(tab)
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/repl.py", line 770, in complete
	self.matches_iter.update(
  File "/home/lukik/anaconda3/envs/ds_env/lib/python3.9/site-packages/bpython/repl.py", line 356, in update
	self.start, self.end, self.current_word = self.completer.locate(
TypeError: cannot unpack non-iterable NoneType object

```

The environment is built using anaconda.
- OS: Ubuntu 22
- Python: 3.9.16
- BPython: 0.20.1
- numpy: 1.23.5

I guess its starting to unpack before the full entry is typed in?

**Comment by sebastinas** [comment]
This issue is fixed in bpython 0.22

**Comment by lukik** [comment]
Upgraded and it works! 

Thanks.
